xml/customFields_install.xml

```diff
@@ -0,0 +1,108 @@
+      <title>External Direct Debit Mandate information</title>
+      <extends>ContributionRecur</extends>
+      <style>Inline</style>
+      <collapse_display>0</collapse_display>
+      <weight>100</weight>
+      <is_active>1</is_active>
+      <table_name>civicrm_value_external_dd_mandate_information</table_name>
+      <is_multiple>0</is_multiple>
+      <collapse_adv_display>0</collapse_adv_display>
+      <is_reserved>0</is_reserved>
+    </CustomGroup>
+  </CustomGroups>
+  <CustomFields>
+    <CustomField>
+      <name>mandate_id</name>
+      <label>Mandate Id</label>
+      <data_type>String</data_type>
+      <html_type>Text</html_type>
+      <is_required>0</is_required>
+      <is_searchable>1</is_searchable>
+      <is_search_range>0</is_search_range>
+      <weight>1</weight>
+      <is_active>1</is_active>
+      <is_view>0</is_view>
+      <text_length>255</text_length>
+      <note_columns>60</note_columns>
+      <note_rows>4</note_rows>
+      <column_name>mandate_id</column_name>
+      <custom_group_name>external_direct_debit_mandate_information</custom_group_name>
+      <in_selector>1</in_selector>
+    </CustomField>
+    <CustomField>
+      <name>mandate_status</name>
+      <label>Mandate Status</label>
+      <data_type>Int</data_type>
+      <html_type>Select</html_type>
+      <is_required>0</is_required>
+      <is_searchable>1</is_searchable>
+      <is_search_range>0</is_search_range>
+      <is_active>1</is_active>
+      <is_view>0</is_view>
+      <text_length>3</text_length>
+      <note_columns>60</note_columns>
+      <note_rows>4</note_rows>
+      <column_name>mandate_status</column_name>
+      <custom_group_name>external_direct_debit_mandate_information</custom_group_name>
+      <option_group_name>automateddirectdebit_mandate_status</option_group_name>
+    </CustomField>
+    <CustomField>
+      <name>next_available_payment_date</name>
+      <label>Next Available Payment Date</label>
+      <data_type>Date</data_type>
+      <html_type>Select Date</html_type>
+      <is_required>0</is_required>
+      <is_searchable>1</is_searchable>
+      <is_search_range>1</is_search_range>
+      <weight>2</weight>
+      <is_active>1</is_active>
+      <is_view>0</is_view>
+      <text_length>255</text_length>
+      <date_format>dd/mm/yy</date_format>
+      <note_columns>60</note_columns>
+      <note_rows>4</note_rows>
+      <column_name>next_available_payment_date</column_name>
+      <custom_group_name>external_direct_debit_mandate_information</custom_group_name>
+      <in_selector>1</in_selector>
+    </CustomField>
+  </CustomFields>
+  <OptionGroups>
+    <OptionGroup>
+      <name>automateddirectdebit_mandate_status</name>
+      <title>Mandate Status</title>
+      <data_type>Int</data_type>
+      <is_reserved>0</is_reserved>
+      <is_active>1</is_active>
+      <is_locked>1</is_locked>
+    </OptionGroup>
+  </OptionGroups>
+  <OptionValues>
```

Comment:
Comment only.

Just so you know, GoCardless have list of mandate status as indicates in this link. https://hub.gocardless.com/s/article/Mandate-statuses?language=en_GB We probably need to map them, so we need to agree that GoCarldess extension should return to the caller. 

Stripe has only three statuses, though.  `active`, `inactive `and pending. https://stripe.com/docs/api/mandates/object#mandate_object-status

Reply:
ah interesting, it seems to me Stripe is opting for simpler approach by minimizing the statuses to just 3, which I assume are the most important ones to manage mandates,  and that we probably don't need to sync all the statuses from GoCardless into Civi, and instead map GoCardless statuses into one of these 3, so:

- "Created" + "Pending Submission" + "Submitted" -> "Pending" 
- "Active" -> "Active"
- "Failed" + "Cancelled" + "Expired" + "Replaced" ..etc -> "Inactive"

because I really don't think we need to manage all the statuses that GoCardless provides and they seems to be redundant for our use case. What do you think ?



If you disagree,  the alternative would be to make the status field a text input instead, and to remove the option group as well that is linked to it, and in each payment processor extension we can define an enumeration that defines the available statuses for each payment processor.

Reply:
@omarabuhussein  I think we just keep it simple, Created, Active and Failed as our job will need to check this status.

Reply:
cool, I will add another statuses "pending" to the available options values.